packages-exp/auth-exp/src/core/providers/phone.ts

```diff
@@ -58,11 +61,27 @@ export class PhoneAuthProvider implements externs.PhoneAuthProvider {
     );
   }
 
-  static _credentialFromResult(
+  static credentialFromResult(
     userCredential: externs.UserCredential
   ): externs.AuthCredential | null {
-    void userCredential;
-    return debugFail('not implemented');
+    const credential = userCredential as UserCredential;
+    assert(
+      credential._tokenResponse,
+      credential.user.auth.name,
+      AuthErrorCode.ARGUMENT_ERROR
+    );
+    const {
+      phoneNumber,
+      temporaryProof
+    } = credential._tokenResponse as SignInWithPhoneNumberResponse;
+    if (phoneNumber && temporaryProof) {
+      return PhoneAuthCredential._fromTokenResponse(
```

Comment:
should we just change the signature here to take the response instead of the two parts?

Reply:
I don't think so. We have to break into those properties here anyway to know whether or not to fail with argument error